library/core/tests/iter/adapters/skip.rs

```diff
@@ -190,3 +192,9 @@ fn test_skip_nth_back() {
     it.by_ref().skip(2).nth_back(10);
     assert_eq!(it.next_back(), Some(&1));
 }
+
+#[test]
+fn test_skip_non_fused() {
+    let non_fused = Unfuse::new(0..10);
+    let _ = non_fused.skip(20).next();
```

Comment:
nit: since it's not obvious where the validation is in here, could you add a little note?  Maybe something like
```suggestion
    let _ = non_fused.skip(20).next(); // `Unfuse` panics if the implementation is uncorrect
```

Reply:
You're right! I've added a note